app/controller/button/TracingMixin.js

```diff
@@ -0,0 +1,275 @@
+/**
+ * A mixin to add tracing functionality to a drawing tool.
+ *
+ * The basic functionality is taken from the official OpenLayers example:
+ * https://openlayers.org/en/latest/examples/tracing.html
+ *
+ * @class CpsiMapview.controller.button.TracingMixin
+ */
+Ext.define('CpsiMapview.controller.button.TracingMixin', {
+    extend: 'Ext.Mixin',
+
+    requires: [
+        'BasiGX.util.Map',
+        'CpsiMapview.util.Tracing'
+    ],
+
+    /**
+     * The style of the preview line during tracing.
+     */
+    previewStyle: new ol.style.Style({
+        stroke: new ol.style.Stroke({
+            color: 'rgba(255, 0, 0, 1)',
+            width: 10,
+        })
+    }),
+
+    /**
+     * Enhances drawing functionality by adding tracing to it.
+     *
+     * Needs to have access to the variable 'editingIsActive' of the parent component.
+     *
+     * @param {String[]} tracingLayerKeys The keys of the layers to trace
+     */
+    initTracing: function (tracingLayerKeys) {
+        var me = this;
+
+        if (me.getView()) {
+            me.getView().fireEvent('tracingstart');
+        }
+
+        // get tracing layers
+        me.tracingLayers = [];
+        Ext.each(tracingLayerKeys, function (key) {
+            var foundLayers = BasiGX.util.Layer.getLayersBy('layerKey', key);
```

Comment:
Is there a helper variable (maybe in BasiGX) that contains the name of the identifying property? If so, it might be better to use that variable instead of the hardcoded value `layerKey`.

Reply:
good point. I cannot find a constant defining the name of `layerKey` value. At other places in the code the layers are searched this way as well.
@geographika do you maybe know if something like this exists in CPSI?
otherwise this could be done in future.

Reply:
We give all layers a `layerKey` property, and use this approach across the project for configuring and finding layers (set in default.json) so I think this is fine. I'm not sure what other variable available that could be used?

Reply:
Apparently the current approach has worked fine so far. I think this was just an idea for improvement. Maybe the name of the layerKey could be stored globally (or in the config file) , and there could be a function like `BasiGX.util.Layer.getLayersByKey(key)`. But this is just syntactic sugar.